src/components/app/searchbox/search-box.jsx

```diff
@@ -0,0 +1,30 @@
+import React, { Component } from 'react'
+import { observer, inject } from 'mobx-react'
+
+@inject('artistsStore', 'artistInfoStore')
+@observer
+class SearchBox extends Component {
+  onInputChange = event => {
+    const { artistsStore, artistInfoStore } = this.props
+
+    const searchedArtist = event.target.value
+    artistsStore.getArtist(searchedArtist)
+    artistInfoStore.updateDisplayInfo(false)
+  }
+
+  render() {
+    return (
+      <div className="sb__input-container">
+        <input
+          className="sb__artist-input"
+          type="text"
+          name="artist"
+          value={this.searchedArtist}
```

Comment:
Hmm tutaj też cieżko połapać się o co chodzi. Skąd się bierze this.searchedArtist? I call `updateDispalyInfo` też jest niemożliwy do rozszyfrowania bez znajomości `InfoStore`.

Reply:
No w jedynej funkcji ktora tam byla byl zmieniany hehe. Ale dodalem `state`, w sumie powinno byc czytelniej.

Co do updateDisplayInfo
![image](https://user-images.githubusercontent.com/30021520/53290061-4f396500-379f-11e9-872a-62af63909912.png)
tego staram sie trzymac, ze logika jest w storesach, wiec _musisz_ je znac, zeby zrozumiec componenty i to byl zamierzony cel. Chyba ze zle rozumiem cos w mobx. ;p

Reply:
Imho źle :P Keep your business logic in stores dla mnie oznacza, że komponent nie powinien wiedzieć nic o business logic, bo jak coś wie to znaczy, że business logic nie jest tylko w store :P I jak musisz znać business logic, żeby zrozumieć komponent to znaczy, że masz pomieszane business logic i UI/komponenty.

Wiadome, że jak jest komponent SearchBox to on musi się połączyć ze Store'm i wiedzieć, że jest metoda getArtists albo updateSearchedArtist w zależności od podejścia ale imho to wszystko. Nie powinien zmieniać `displayInfo`, bo nie będziesz mógł go wykorzystać w sytuacji, w której nie będziesz chciał zmieniać `displayInfo`

Reply:
No moge w sumie dodac jakis `@computed` ze jesli mam searchedArtist to zmien na true w samym storze. Zapisze sobie do refactoru componentow 😅

Reply:
Ok ;)